src/Agent.php

```diff
@@ -681,13 +681,99 @@ public static function getIcon()
         return "ti ti-robot";
     }
 
+    /**
+     * Clean and do other defined actions on an agent or related item
+     * @param Agent $agent The agent to clean
+     * @param array $config Array containing the agent clean config
+     * @return bool True if successful
+     */
+    private static function cleanAgent(Agent $agent, array $config): bool
+    {
+        global $PLUGIN_HOOKS;
+
+        // Actions to apply to the agent or item
+        /** @phpstan-var array{item_action: bool, callback: callable(?Agent, array, ?CommonDBTM):bool}[] $actions_to_apply */
+        $actions_to_apply = [];
+        $need_item = false;
+        if (isset($config['stale_agents_status']) && $config['stale_agents_status'] >= 0) {
+            // 0 = none, -1 or other negative = no change
+            $need_item = true;
+            $actions_to_apply[] = [
+                'item_action' => true,
+                'callback' => static function (?Agent $agent, array $config, ?CommonDBTM $item): bool {
+                    return $item !== null && $item->update([
+                        'id' => $item->fields['id'],
+                        'states_id' => $config['stale_agents_status']
+                    ]);
+                }
+            ];
+        }
+        $plugin_actions = $PLUGIN_HOOKS[Hooks::STALE_AGENT_CONFIG] ?? [];
+        /**
+         * @var string $plugin
+         * @phpstan-var array{label: string, item_action: boolean, render_callback: callable, action_callback: callable}[] $actions
+         */
+        foreach ($plugin_actions as $plugin => $actions) {
+            if (is_array($actions) && \Plugin::isPluginActive($plugin)) {
+                foreach ($actions as $action) {
```

Comment:
A check on hook declaration would probably be more secure.
```suggestion
                foreach ($actions as $action) {
                    if (!is_callable($action['action_callback'] ?? null)) {
                        trigger_error(
                            sprintf('Invalid plugin "%s" action callback for "%s" hook.', $plugin, Hooks::STALE_AGENT_CONFIG),
                            E_USER_WARNING
                        );
                        continue;
                    }
```